src/prefect/deployments/runner.py

```diff
@@ -437,7 +443,7 @@ def from_flow(
         rrule: Optional[Union[Iterable[str], str]] = None,
         paused: Optional[bool] = None,
         schedules: Optional["FlexibleScheduleList"] = None,
-        concurrency_limit: Optional[int] = None,
+        concurrency_limit: Optional[Union[int, ConcurrencyLimitConfig]] = None,
```

Comment:
I think typing unions that can also be `None` like this is more readable.
```suggestion
        concurrency_limit: Union[int, ConcurrencyLimitConfig, None] = None,
```